larq_compute_engine/mlir/python/converter.py

```diff
@@ -57,6 +61,119 @@ def _contains_training_quant_op(graph_def):
     return False
 
 
+def _validate_options(
+    *,
+    inference_input_type=None,
+    inference_output_type=None,
+    target=None,
+    experimental_default_int8_range=None,
+):
+    if inference_input_type not in (tf.float32, tf.int8):
+        raise ValueError(
+            "Expected `inference_input_type` to be either `tf.float32` or `tf.int8`, "
+            f"got {inference_input_type}."
+        )
+    if inference_output_type not in (tf.float32, tf.int8):
+        raise ValueError(
+            "Expected `inference_output_type` to be either `tf.float32` or `tf.int8`, "
+            f"got {inference_output_type}."
+        )
+    if target not in ("arm", "xcore"):
+        raise ValueError(f'Expected `target` to be "arm" or "xcore", but got {target}.')
+
+    if not tf.executing_eagerly():
+        raise RuntimeError(
+            "Graph mode is not supported. Please enable eager execution using "
+            "tf.enable_eager_execution() when using TensorFlow 1.x"
+        )
+    if experimental_default_int8_range:
+        warnings.warn(
+            "Using `experimental_default_int8_range` as fallback quantization stats. "
+            "This should only be used for latency tests."
+        )
+
+
+def convert_saved_model(
+    saved_model_dir: Union[str, os.PathLike],
+    *,  # Require remaining arguments to be keyword-only.
+    inference_input_type: tf.DType = tf.float32,
+    inference_output_type: tf.DType = tf.float32,
+    target: str = "arm",
+    experimental_default_int8_range: Optional[Tuple[float, float]] = None,
+    experimental_enable_bitpacked_activations: bool = False,
+) -> bytes:
+    """Converts a SavedModel to TFLite flatbuffer.
```

Comment:
I'm never sure how we should refer to things as being TensorFlow versus being Larq/LCE.

On the one hand, it's accurate because it _is_ a TFLite flatbuffer, just with extra custom ops. On the other hand, I feel like it's weird for a user to use the `larq_compute_engine` package and have method docstrings that refer to TensorFlow but not Larq or LCE.

I'm probably overthinking this though - how do you feel about it?

Reply:
This is pretty much copied from the `convert_keras_model` function, but I'm happy to rephrase it. Not sure how we refer to it in the guides though.

Reply:
Yeah me neither, I think that's the real issue. Since we are going to be going through and updating our docs for the 0.6 release anyhow, we can maybe have a think about it then and not worry about it now.